gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslator.java

```diff
@@ -0,0 +1,309 @@
+    /**
+     * Creates the translator with a custom {@link TypeTranslator} instance.
+     */
+    public static GolangTranslator of(final String traversalSource, final TypeTranslator typeTranslator) {
+        return new GolangTranslator(traversalSource, typeTranslator);
+    }
+
+    @Override
+    public String getTraversalSource() {
+        return this.traversalSource;
+    }
+
+    @Override
+    public Script translate(final Bytecode bytecode) {
+        return typeTranslator.apply(traversalSource, bytecode);
+    }
+
+    @Override
+    public String getTargetLanguage() {
+        return "gremlin-go";
+    }
+
+    @Override
+    public String toString() {
+        return StringFactory.translatorString(this);
+    }
+
+    ///////
+
+    /**
+     * Performs standard type translation for the TinkerPop types to Go.
+     */
+    public static class DefaultTypeTranslator extends AbstractTypeTranslator {
+
+        public DefaultTypeTranslator(final boolean withParameters) {
+            super(withParameters);
+        }
+
+        @Override
+        protected String getNullSyntax() {
+            return "nil";
+        }
+
+        @Override
+        protected String getSyntax(final String o) {
+            return "\"" + StringEscapeUtils.escapeJava(o) + "\"";
+        }
+
+        @Override
+        protected String getSyntax(final Boolean o) {
+            return o.toString();
+        }
+        @Override
+        protected String getSyntax(final Date o) {
+            return "time.Unix(" + o.getTime() + ", 0)";
+        }
+
+        @Override
+        protected String getSyntax(final Timestamp o) {
+            return "time.Unix(" + o.getTime() + ", 0)";
+        }
+
+        @Override
+        protected String getSyntax(final UUID o) {
+            return "uuid.MustParse(\"" + o.toString() + "\")";
+        }
+
+        @Override
+        protected String getSyntax(final Lambda o) {
+            // TODO: AN-1037 Lambda support in Gremlin-Go
+            // return "gremlingo.GroovyLambda(\"" + StringEscapeUtils.escapeEcmaScript(o.getLambdaScript().trim()) + "\")";
+            throw new NotImplementedException("Lambda translation not currently supported");
+        }
+
+        @Override
+        protected String getSyntax(final Number o) {
+                return o.toString();
+        }
+
+        @Override
+        protected String getSyntax(final SackFunctions.Barrier o) {
+            return "Barrier." + resolveSymbol(o.toString());
+        }
+
+        @Override
+        protected String getSyntax(final VertexProperty.Cardinality o) {
+            return "Cardinality." + resolveSymbol(o.toString());
+        }
+
+        @Override
+        protected String getSyntax(final TraversalOptionParent.Pick o) {
+            return "Pick." + resolveSymbol(o.toString());
+        }
+
+        @Override
+        protected Script produceScript(final Set<?> o) {
+            // TODO: AN-1032 Change this when Set type is added in Gremlin-Go
```

Comment:
Sort of annoying, but can you make a separate ticket to address this? Rithin's Set implementation is already in review and won't have a chance to update this before merge.